antarest/study/business/aggregator_management.py

```
@@ -144,20 +227,97 @@ def _gather_all_files_to_consider(self) -> t.Sequence[Path]:
+        return df
+
+    def _process_df(self, file_path: Path, is_details: bool) -> pd.DataFrame:
+        if is_details:
+            un_normalized_df = self._parse_output_file(file_path, normalize=False)
+            df_len = len(un_normalized_df)
+            cluster_dummy_product_cols = sorted(
+                set([(x[CLUSTER_ID_COMPONENT], x[DUMMY_COMPONENT]) for x in un_normalized_df.columns])
+            )
+            actual_cols = sorted(set(un_normalized_df.columns.map(lambda x: x[ACTUAL_COLUMN_COMPONENT])))
+            new_obj: t.Dict[str, t.Any] = {k: [] for k in actual_cols}
+            new_obj[CLUSTER_ID_COL] = []
+            new_obj[TIME_ID_COL] = []
+            for cluster_id, dummy_component in cluster_dummy_product_cols:
+                for actual_col in actual_cols:
+                    col_values = un_normalized_df[(cluster_id, actual_col, dummy_component)].tolist()  # type: ignore
+                    new_obj[actual_col] += col_values
+                new_obj[CLUSTER_ID_COL] += [cluster_id for _ in range(df_len)]
+                new_obj[TIME_ID_COL] += list(range(1, df_len + 1))
+
+            prod_col = _infer_production_column(actual_cols)
+            if prod_col is not None:
+                new_obj[PRODUCTION_COLUMN_NAME] = new_obj.pop(prod_col)
+                actual_cols.remove(prod_col)
+
+            add_prod = [PRODUCTION_COLUMN_NAME] if prod_col is not None else []
+            columns_order = [CLUSTER_ID_COL, TIME_ID_COL] + add_prod + list(actual_cols)
+            df = pd.DataFrame(new_obj).reindex(columns=columns_order).sort_values(by=[TIME_ID_COL, CLUSTER_ID_COL])
+            df.index = pd.Index(list(range(1, len(df) + 1)))
+
+            return df
+
+        else:
+            return self._parse_output_file(file_path)
+
     def _build_dataframe(self, files: t.Sequence[Path], horizon: int) -> pd.DataFrame:
+        assert self.mc_root in [MCRoot.MC_IND, MCRoot.MC_ALL], f"Unknown Monte Carlo root: {self.mc_root}"
```

Not sure assertion is the way to go, you should probably raise an Exception as assertions are removed when optimization are requested from Python

---

I will handle in an other way